packages/discord.js/src/structures/UncachedGuildMember.js

```diff
@@ -0,0 +1,328 @@
+   * @name UncachedGuildMember#roles
+   * @type {Snowflake[]}
+   * @deprecated Use {@link UncachedGuildMember#roleIds} instead
+   */
+  get roles() {
+    return this.roleIds;
+  }
+
+  /**
+   * Whether this member is currently timed out
+   * @returns {boolean}
+   */
+  isCommunicationDisabled() {
+    return this.communicationDisabledUntilTimestamp > Date.now();
+  }
+
+  /**
+   * Creates a DM channel between the client and this member.
+   * @param {boolean} [force=false] Whether to skip the cache check and request the API
+   * @returns {Promise<DMChannel>}
+   */
+  createDM(force = false) {
+    return this.user.createDM(force);
+  }
+
+  /**
+   * Deletes a DM channel (if one exists) between the client and the member. Resolves with the channel if successful.
+   * @returns {Promise<DMChannel>}
+   */
+  deleteDM() {
+    return this.user.deleteDM();
+  }
+
+  /**
+   * Whether this guild member equals another guild member. It compares all properties, so for most
+   * comparison it is advisable to just compare `member.id === member2.id` as it is significantly faster
+   * and is often what most users need.
+   * @param {UncachedGuildMember} member The member to compare with
+   * @returns {boolean}
+   */
+  equals(member) {
+    return (
+      member instanceof this.constructor &&
+      this.id === member.id &&
+      this.partial === member.partial &&
+      this.guildId === member.guildId &&
+      this.joinedTimestamp === member.joinedTimestamp &&
+      this.nickname === member.nickname &&
+      this.avatar === member.avatar &&
+      this.pending === member.pending &&
+      this.communicationDisabledUntilTimestamp === member.communicationDisabledUntilTimestamp &&
+      this.parsedFlags.bitfield === member.parsedFlags.bitfield &&
+      (this.roleIds === member.roleIds ||
+        (this.roleIds.length === member.roleIds.length && this.roleIds.every((role, i) => role === member.roleIds[i])))
+    );
+  }
+
+  /**
+   * When concatenated with a string, this automatically returns the user's mention
+   * instead of the UncachedGuildMember object.
+   * @returns {string}
+   * @example
+   * // Logs: Hello from <@123456789012345678>!
+   * console.log(`Hello from ${member}!`);
+   */
+  toString() {
+    return this.user.toString();
+  }
+
+  toJSON() {
+    const json = super.toJSON({
+      guildId: true,
+      user: 'userId',
+      displayName: true,
+      roles: true,
+    });
+    json.avatarURL = this.avatarURL();
+    json.displayAvatarURL = this.displayAvatarURL();
+    return json;
+  }
+}
+
+/**
+ * Sends a message to this user.
+ * @method send
+ * @memberof UncachedGuildMember
+ * @instance
+ * @param {string|MessagePayload|MessageCreateOptions} options The options to provide
+ * @returns {Promise<Message>}
+ * @example
+ * // Send a direct message
+ * UncachedGuildMember.send('Hello!')
+ *   .then(message => console.log(`Sent message: ${message.content} to ${guildMember.displayName}`))
```

Comment:
```suggestion
 *   .then(message => console.log(`Sent message: ${message.content} to ${uncachedGuildMember.displayName}`))
```
Feels a bit silly but eh